app/jobs/ingest_yaml_job.rb

```diff
@@ -115,4 +118,19 @@ def yaml_to_repo_map
     def thumbnail_path
       @thumbnail_path ||= @yaml[:thumbnail_path]
     end
+
+    def advance_state(resource, final_state)
+      until resource.state == final_state
+        permitted = StateWorkflow.new(resource.state).valid_transitions.map(&:to_s)
+        if permitted.include? final_state
+          resource.state = final_state
+        elsif permitted.size == 1
+          resource.state = permitted.first
+        else
+          raise "Unknown state transition path from #{resource.state} to #{final_state} through #{permitted.join(',')}"
```

Comment:
"Ambiguous" might be better than "Unknown" when permitted.size > 1, since it means that there are multiple paths between resource.state and final_state.  Is it possible for permitted.size to be <1?  That situation would be "unknown", since it means there is no path between resource.state and final_state.

Is there some way we can check the whole state graph once when it is loaded and kick it out with an error if it is nondeterministic?